src/index/base.cpp

```diff
@@ -66,7 +66,16 @@ bool BaseIndex::Init()
     } else {
         m_best_block_index = m_chainstate->m_blockman.FindForkInGlobalIndex(active_chain, locator);
     }
+
+    // Note: this will latch to true immediately if the user starts up with an empty
+    // datadir and an index enabled. If this is the case, indexation will happen solely
+    // via `BlockConnected` signals until, possibly, the next restart.
     m_synced = m_best_block_index.load() == active_chain.Tip();
+    if (!m_best_block_index && !active_chain.Tip()) {
+        LogPrintf("%s: %s considered fully synced due to empty chain\n",
```

Comment:
I also think that there shouldn't be a `LogPrintf` here: At this point in time, we only use this information for skipping the prune violation checks, we aren't "officially" synced.
For signaling to a user that an index is synced, we already have a `LogPrintf` in [`ThreadSync`](https://github.com/bitcoin/bitcoin/blob/ae005a647ffa1d457c5c7a0528cb29f3b1937b96/src/index/base.cpp#L195-L199): It currently says "[indexname] is enabled" in case of an empty chain, and "[indexname] is enabled at height X" otherwise - we could to add an "due to an empty chain" here.
But two messages for the same thing would only be confusing in my opinion.

Reply:
@mzumsande, maybe you could be specific about what is confusing? Do you only think this is redundant, or do you think there is something misleading or contradictory about these messages?

If I were adding a log message here I'd probably say "Chain height is 0. Index is synced." but I don't know if that helps. 

Message in the PR seems basically ok to me. (I would just drop "fully" because no index will be fully synced until the last bitcoin block is produced)

Reply:
The log is 
```
Init: txindex considered fully synced due to empty chain
txindex thread start
txindex is enabled
txindex thread exit
```
(potentially x3 for each index). 
I'd find it a bit confusing why we'd need to start a thread to sync an index right after saying that the index is fully synced. So I think it could be simpler:
```
txindex thread start
txindex is enabled due to an empty chain
txindex thread exit
```

Reply:
> I'd find it a bit confusing why we'd need to start a thread to sync an index right after saying that the index is fully synced. 

I see. You don't think the log message gives a misleading impression. You think the log message gives a correct impression that the code is doing something stupid, and it's confusing why the stupid thing would be happening. Fair enough!

Reply:
I agree with @mzumsande that this is unnecessary. It just doesn't seem like the right place for this kind of information. The information that the code is something stupid seems fine as just a code comment while the logs are also viewed by non-developers who might be confused.

Reply:
I've removed this log; now the entire change is doc-only.

Reply:
Doesn't look like anything has changed, did you push the wrong commit?

Reply:
Oops, thanks @mzumsande.